src/main/resources/migration/changelog-v5.4.0.xml

```diff
@@ -0,0 +1,253 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog
+        objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+            http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+            http://www.liquibase.org/xml/ns/dbchangelog
+            http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
+    <changeSet id="v5.4.0-1" author="sahibamittal">
+        <modifyDataType  columnName="COMMENTER" newDataType="TEXT" tableName="ANALYSISCOMMENT"/>
```

Comment:
Should we also convert the types for fields holding enum values (`ANALYSIS.JUSTIFICATION` etc.)?

As we have proper value `CHECK`s now, I don't think we need the length constraint of `VARCHAR(N)` anymore either.

Reply:
It is already added in the changelog : https://github.com/DependencyTrack/hyades-apiserver/blob/5d5b770da5f88b7abcbb3f978186764b9d42058d/src/main/resources/migration/changelog-v5.4.0.xml#L160

Reply:
Yes, the `CHECK` constraint is, but not the conversion from `VARCHAR(N)` to `TEXT`.

Reply:
oh you mean modify data types for these as well, yes, will add it for all enum properties being modified 👍🏼

Reply:
We're still missing a few conversions here. You can validate by running this SQL:

```sql
select
  table_name, column_name, data_type
from
  information_schema.columns
where
  -- exclude postgres-internal tables
  table_schema = 'public'
  -- exclude alpine tables, can't change those
  and table_name not in ('APIKEY', 'CONFIGPROPERTY', 'EVENTSERVICELOG', 'INSTALLEDUPGRADES', 'LDAPUSER', 'PERMISSION', 'MANAGEDUSER', 'MAPPEDLDAPGROUP', 'OIDCGROUP', 'OIDCUSER', 'SCHEMAVERSION', 'TEAM')
  -- exclude liquibase tables, can't change those
  and table_name not in ('databasechangelog', 'databasechangeloglock')
  -- exclude shedlock tables, can't change those
  and table_name != 'shedlock'
  -- exclude uuid fields, those should stay length-restricted
  and column_name != 'UUID'
  and data_type = 'character varying';
```

After applying this PR, I get:

| table\_name | column\_name | data\_type |
| :--- | :--- | :--- |
| INTEGRITY\_META\_COMPONENT | MD5 | character varying |
| INTEGRITY\_META\_COMPONENT | SHA1 | character varying |
| INTEGRITY\_META\_COMPONENT | SHA256 | character varying |
| INTEGRITY\_META\_COMPONENT | SHA512 | character varying |
| INTEGRITY\_META\_COMPONENT | STATUS | character varying |
| COMPONENT | BLAKE2B\_256 | character varying |
| COMPONENT | BLAKE2B\_384 | character varying |
| COMPONENT | BLAKE2B\_512 | character varying |
| COMPONENT | BLAKE3 | character varying |
| COMPONENT | CLASSIFIER | character varying |
| COMPONENT | MD5 | character varying |
| COMPONENT | SHA1 | character varying |
| COMPONENT | SHA\_256 | character varying |
| COMPONENT | SHA\_384 | character varying |
| COMPONENT | SHA3\_256 | character varying |
| COMPONENT | SHA3\_384 | character varying |
| COMPONENT | SHA3\_512 | character varying |
| COMPONENT | SHA\_512 | character varying |
| WORKFLOW\_STATE | STATUS | character varying |
| WORKFLOW\_STATE | STEP | character varying |
| WORKFLOW\_STATE | TOKEN | character varying |
| VULNERABILITY\_POLICY\_BUNDLE | HASH | character varying |
| PROJECT | CLASSIFIER | character varying |
| ANALYSIS | CVSSV2VECTOR | character varying |
| ANALYSIS | OWASPVECTOR | character varying |
| ANALYSIS | CVSSV3VECTOR | character varying |
| ANALYSIS | SEVERITY | character varying |
| VULNERABILITYSCAN | TARGET\_IDENTIFIER | character varying |

Hashes, vectors, and `WORKFLOW_STATE.TOKEN` + `VULNERABILITYSCAN.TARGET_IDENTIFIER` (both UUIDs) are OK to leave length-restricted, as discussed.

However, `ANALYSIS.SEVERITY`, `COMPONENT.CLASSIFIER`, `PROJECT.CLASSIFIER`, `WORKFLOW_STATE.STATUS`, and `WORKFLOW_STATE.STEP` should be converted just like other enum columns. `CHECK`s already exist for them.